packages/components/src/box-control/vertical-horizontal-input-controls.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Internal dependencies
+ */
+import UnitControl from './unit-control';
+import { LABELS } from './utils';
+import { Layout } from './styles/box-control-styles';
+
+const noop = () => {};
+const groupedSides = [ 'vertical', 'horizontal' ];
+
+export default function VerticalHorizontalInputControls( {
+	onChange = noop,
```

Comment:
As @ciampo pointed out, we've been removing a bunch of unnecessary Lodash usage in other repositories. 

Something I can bring in here from my experience with that is my observations a pattern that I also notice here. I'm talking about using `noop` as default for event handler functions. Is that default value really necessary? Can't we just use optional chaining when invoking them: `onChange.?()`? Also, another thing to point out is that for example in `createHandleOnHoverOn`, if `onHoverOn` is `noop`, we'll be executing all of the code inside in vain. If we didn't pass a default function but had `null` or `undefined` instead, we could check if a function exists and bail early, avoiding the execution of unnecessary logic.

What do you think?

Reply:
Good points @tyxla! Here, for consistency, I was mostly copying the pattern that I saw used in the two comparable Controls functions in this component, without rethinking the way it was going about managing its event handlers. So thanks for jumping in and raising it! 

In terms of readability and to avoid making mistakes, I think I prefer adding an explicit `if ( ! onFocus ) { return; }` check at the beginning of each handler function instead of using optional chaining. Without TypeScript for type checking our props, I could imagine it being easy to accidentally add a call to `onHoverOff()` and forget to add the `?.()` and wind up calling an undefined variable? I suppose `noop` had been used as a catch-all to prevent this? The downside of the early checks is that it adds additional lines of code, but the clear up-side is that as you say, we avoid executing code unnecessarily.

So, of the listed options, I think I quite like the option of bailing early in the event handlers if the handler doesn't exist. I've just updated this PR quickly before I finish for the day, but happy to make further changes tomorrow, too, or go in a different direction if folks have preferences! 🙂

Reply:
I'm perfectly happy with the early `return`. It also might be more familiar to folks who are not used to using optional chaining. It might be more LoC, but if we stack the effect of not executing code unnecessarily for multiple components rendered multiple times on the same page, it might end up having some positive effect on the performance.

Thanks for accommodating my feedback in such a timely manner 🙌